vmdb/app/models/mixins/web_server_worker_mixin.rb

```diff
@@ -75,6 +75,9 @@ def self.sync_workers
       result  = { :adds => [], :deletes => [] }
       ports = self.all_ports_in_use
 
+      # TODO: This tracking of adds/deletes of pids and ports is not DRY
```

Comment:
DRY with `ports / all_ports_in_use` ?

you could use `reserve_port(ports + ports_hash[:adds] - ports_hash[:remove])`

But I think it is fine.